File: 1146-greatest-common-divisor-of-strings/greatest-common-divisor-of-strings.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        m,n = len(str1),len(str2)
        min_r = min(m,n)
        for r in range(min_r,0,-1):
            if m%r ==0 and n%r ==0:
                curr = str1[:r]
                if curr*(m//r) ==str1 and curr*(n//r) ==str2:
                    return curr
        return ""

File: 1146-greatest-common-divisor-of-strings/greatest-common-divisor-of-strings.ts
function gcdOfStrings(str1: string, str2: string): string {
  const [m,n] = [str1.length, str2.length]
  const min_r = Math.min(m,n)
  for (let k = min_r; k>=0;k--){
    if ( m%k ==0 && n%k == 0){
        let curr = str1.slice(0,k)
        if (curr.repeat(Math.floor(m/k))==str1 && curr.repeat(Math.floor(n/k))==str2) {
            return curr
        }
    }
  }
  return ""
};